### ISSUE TITLE
Add support for ingress_application_gateway block  in azurerm_kubernetes_cluster resource

### ISSUE THREAD

**Issue by mpalacios-uhg** [aztfmod/terraform-azurerm-caf#733]
Judging by the existing code, it looks as if something like the following will fit the bill

Module:
   https://github.com/aztfmod/terraform-azurerm-caf/blob/master/modules/compute/aks/aks.tf

Code: 
```
...

resource "azurerm_kubernetes_cluster" "aks" {
   dynamic "addon_profile" {

   ...

      dynamic "ingress_application_gateway" {
         for_each = try(var.settings.addon_profile.ingress_application_gateway[*], {})

         content {
            enabled      = http_application_routing.value.enabled
            gateway_id   = try(ingress_application_gateway.value.gateway_id, null)
            gateway_name = try(ingress_application_gateway.value.gateway_name, null)
            subnet_cidr  = try(ingress_application_gateway.value.subnet_cidr, null)
            subnet_id    = try(ingress_application_gateway.value.subnet_id, null)
         }
      }
   }

...

}

...
```

**Comment by arnaudlh** [comment]
hi @mpalacios-uhg, this is under @hieumoscow's PR which will be merged in 5.5.0. thnks!

**Comment by mpalacios-uhg** [comment]
Hi @arnaudlh thanks for letting me know. I've notified my successors at my previous team to watch this issue.

**Comment by arnaudlh** [comment]
Included in 5.5.0 and backported in 5.4.8 also!
